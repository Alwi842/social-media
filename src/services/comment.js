import axios from "axios";
export const getCommentById = async (id) => {
  try {
    const response = await axios.get(`https://jsonplaceholder.typicode.com/comments/?postId=${id}`);
    return response.data;
  } catch (err) {
    throw new Error("failed to fetch data: ", err);
  }
};
