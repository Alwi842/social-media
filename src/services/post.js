import axios from "axios";
export const getPostById = async (id) => {
  try {
    const response = await axios.get(`https://jsonplaceholder.typicode.com/posts/${id}`);
    return response.data;
  } catch (err) {
    throw new Error("failed to fetch data: ", err);
  }
};
