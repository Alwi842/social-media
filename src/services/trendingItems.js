export const getTrendingItems = async () => {
  //fake trending
  const trendingItems = [
    "React 18 Released",
    "JavaScript Tips",
    "Next.js vs Gatsby",
    "Web3 Development",
    "CSS Grid Layout",
    "AI in Web Development",
    "Frontend Frameworks",
    "Node.js Performance",
  ];
  return trendingItems;
};
