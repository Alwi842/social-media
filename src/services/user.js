import axios from "axios";

export const getUserById = async (userId) => {
  try {
    const response = await axios.get(`https://jsonplaceholder.typicode.com/users/${userId}`);

    // Return user data with the avatar
    return {
      [userId]: {
        ...response.data, // Add fetched user data
        avatar: `https://picsum.photos/id/${userId}/100`, // Add avatar
      },
    };
  } catch (error) {
    // Log error and return an empty object or handle the error as needed
    console.error("Error fetching user:", error);
    return {}; // Return an empty object to avoid crashing the app
  }
};
export const getUsers = async () => {
  try {
    const response = await axios.get(`https://jsonplaceholder.typicode.com/users/`);

    // Iterate through the response data and add avatar for each user
    const usersWithAvatars = response.data.map((user) => ({
      ...user, // Spread existing user data
      avatar: `https://picsum.photos/id/${user.id}/100`, // Add avatar based on user id
    }));

    // Return the users with avatar data, using their id as the key
    const usersById = usersWithAvatars.reduce((acc, user) => {
      acc[user.id] = user;
      return acc;
    }, {});

    return usersById;
  } catch (error) {
    // Log error and return an empty object or handle the error as needed
    console.error("Error fetching users:", error);
    return {}; // Return an empty object to avoid crashing the app
  }
};
