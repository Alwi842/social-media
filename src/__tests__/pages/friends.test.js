import { render } from "@testing-library/react";
import Friends from "@/pages/friends";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom";
import { useRouter } from "next/router";
import { getUsers } from "@/services/user";

// Mock Next.js router
jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

// Mock Redux store
const mockStore = configureStore();
const store = mockStore({
  screen: { isMobileScreen: false }, // Control mobile/desktop rendering
});

// Mock `getUsers` service
jest.mock("@/services/user", () => ({
  getUsers: jest.fn(),
}));

describe("Friends Page", () => {
  beforeEach(() => {
    // Mock useRouter behavior
    useRouter.mockReturnValue({
      back: jest.fn(),
    });

    // Mock getUsers to return fake user data
    getUsers.mockResolvedValue([
      { id: 1, name: "John Doe", username: "johndoe", email: "john@example.com", avatar: "/avatar1.png" },
      { id: 2, name: "Jane Smith", username: "janesmith", email: "jane@example.com", avatar: "/avatar2.png" },
    ]);
  });

  it("renders and matches snapshot", async () => {
    const { container } = render(
      <Provider store={store}>
        <Friends
          users={[
            { id: 1, name: "John Doe", username: "johndoe", email: "john@example.com", avatar: "/avatar1.png" },
            { id: 2, name: "Jane Smith", username: "janesmith", email: "jane@example.com", avatar: "/avatar2.png" },
          ]}
        />
      </Provider>
    );

    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});
