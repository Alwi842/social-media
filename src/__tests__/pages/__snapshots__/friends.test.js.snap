// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friends Page renders and matches snapshot 1`] = `
<div>
  <div
    class="flex"
  >
    <div
      class="h-screen bg-gray-900 text-white p-4"
    >
      <div
        class="w-40 h-full bg-gray-900 text-white  space-y-6"
      >
        <div
          class="text-xl font-semibold"
        >
          SocialMedia
        </div>
        <div
          class="space-y-4"
        >
          <a
            class="flex items-center text-lg hover:bg-gray-700 p-2 rounded-lg gap-2"
            href="/home"
          >
            <svg
              fill="#e8eaed"
              height="24px"
              viewBox="0 -960 960 960"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M240-200h120v-240h240v240h120v-360L480-740 240-560v360Zm-80 80v-480l320-240 320 240v480H520v-240h-80v240H160Zm320-350Z"
              />
            </svg>
            Home
          </a>
          <a
            class="flex items-center text-lg hover:bg-gray-700 p-2 rounded-lg gap-2"
            href="/notification"
          >
            <svg
              fill="#e8eaed"
              height="24px"
              viewBox="0 -960 960 960"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M160-200v-80h80v-280q0-83 50-147.5T420-792v-28q0-25 17.5-42.5T480-880q25 0 42.5 17.5T540-820v28q80 20 130 84.5T720-560v280h80v80H160Zm320-300Zm0 420q-33 0-56.5-23.5T400-160h160q0 33-23.5 56.5T480-80ZM320-280h320v-280q0-66-47-113t-113-47q-66 0-113 47t-47 113v280Z"
              />
            </svg>
            Notifications
          </a>
          <a
            class="flex items-center text-lg hover:bg-gray-700 p-2 rounded-lg gap-2"
            href="/friends"
          >
            <svg
              fill="#e8eaed"
              height="24px"
              viewBox="0 -960 960 960"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0-240v-63q0-43 44-70t116-27q13 0 25 .5t23 2.5q-14 21-21 44t-7 48v65H0Zm240 0v-65q0-32 17.5-58.5T307-410q32-20 76.5-30t96.5-10q53 0 97.5 10t76.5 30q32 20 49 46.5t17 58.5v65H240Zm540 0v-65q0-26-6.5-49T754-397q11-2 22.5-2.5t23.5-.5q72 0 116 26.5t44 70.5v63H780Zm-455-80h311q-10-20-55.5-35T480-370q-55 0-100.5 15T325-320ZM160-440q-33 0-56.5-23.5T80-520q0-34 23.5-57t56.5-23q34 0 57 23t23 57q0 33-23 56.5T160-440Zm640 0q-33 0-56.5-23.5T720-520q0-34 23.5-57t56.5-23q34 0 57 23t23 57q0 33-23 56.5T800-440Zm-320-40q-50 0-85-35t-35-85q0-51 35-85.5t85-34.5q51 0 85.5 34.5T600-600q0 50-34.5 85T480-480Zm0-80q17 0 28.5-11.5T520-600q0-17-11.5-28.5T480-640q-17 0-28.5 11.5T440-600q0 17 11.5 28.5T480-560Zm1 240Zm-1-280Z"
              />
            </svg>
            Friends
          </a>
          <a
            class="flex items-center text-lg hover:bg-gray-700 p-2 rounded-lg gap-2"
            href="/settings"
          >
            <svg
              fill="#e8eaed"
              height="24px"
              viewBox="0 -960 960 960"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m370-80-16-128q-13-5-24.5-12T307-235l-119 50L78-375l103-78q-1-7-1-13.5v-27q0-6.5 1-13.5L78-585l110-190 119 50q11-8 23-15t24-12l16-128h220l16 128q13 5 24.5 12t22.5 15l119-50 110 190-103 78q1 7 1 13.5v27q0 6.5-2 13.5l103 78-110 190-118-50q-11 8-23 15t-24 12L590-80H370Zm70-80h79l14-106q31-8 57.5-23.5T639-327l99 41 39-68-86-65q5-14 7-29.5t2-31.5q0-16-2-31.5t-7-29.5l86-65-39-68-99 42q-22-23-48.5-38.5T533-694l-13-106h-79l-14 106q-31 8-57.5 23.5T321-633l-99-41-39 68 86 64q-5 15-7 30t-2 32q0 16 2 31t7 30l-86 65 39 68 99-42q22 23 48.5 38.5T427-266l13 106Zm42-180q58 0 99-41t41-99q0-58-41-99t-99-41q-59 0-99.5 41T342-480q0 58 40.5 99t99.5 41Zm-2-140Z"
              />
            </svg>
            Settings
          </a>
        </div>
      </div>
    </div>
    <div
      class="w-full bg-black min-h-screen flex flex-col text-white "
    >
      <div
        class="w-full p-6 space-y-6 overflow-y-auto"
      >
        <button
          class="flex gap-2"
        >
          <svg
            fill="#e8eaed"
            height="24px"
            viewBox="0 -960 960 960"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M640-200 200-480l440-280v560Zm-80-280Zm0 134v-268L350-480l210 134Z"
            />
          </svg>
          Back
        </button>
        <div
          class="flex justify-center items-center w-full"
        >
          <h1
            class="text-3xl font-semibold text-center"
          >
            Friends
          </h1>
        </div>
        <div
          class="flex-1 "
        >
          <div
            class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-6"
          >
            <div
              class="bg-black border-white p-4 border rounded-lg shadow-md flex flex-col items-center"
            >
              <img
                alt="John Doe's avatar"
                class="rounded-full w-24 h-24 mb-4"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=%2Favatar1.png&w=256&q=75"
                srcset="/_next/image?url=%2Favatar1.png&w=128&q=75 1x, /_next/image?url=%2Favatar1.png&w=256&q=75 2x"
                style="color: transparent;"
                width="100"
              />
              <h2
                class="text-lg font-medium"
              >
                John Doe
              </h2>
              <p
                class="text-sm text-gray-500"
              >
                johndoe
              </p>
              <p
                class="text-sm text-gray-600"
              >
                john@example.com
              </p>
            </div>
            <div
              class="bg-black border-white p-4 border rounded-lg shadow-md flex flex-col items-center"
            >
              <img
                alt="Jane Smith's avatar"
                class="rounded-full w-24 h-24 mb-4"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=%2Favatar2.png&w=256&q=75"
                srcset="/_next/image?url=%2Favatar2.png&w=128&q=75 1x, /_next/image?url=%2Favatar2.png&w=256&q=75 2x"
                style="color: transparent;"
                width="100"
              />
              <h2
                class="text-lg font-medium"
              >
                Jane Smith
              </h2>
              <p
                class="text-sm text-gray-500"
              >
                janesmith
              </p>
              <p
                class="text-sm text-gray-600"
              >
                jane@example.com
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
